components/kubeflow/katib-launcher/src/launch_experiment.py

```diff
@@ -18,123 +18,148 @@
 import json
 import os
 import logging
-import yaml
-import uuid
-import launch_crd
-
-from kubernetes import client as k8s_client
-from kubernetes import config
-
-def yamlOrJsonStr(str):
-    if str == "" or str == None:
-        return None
-    try:
-        return json.loads(str)
-    except:
-        return yaml.safe_load(str)
-
-ExperimentGroup = "kubeflow.org"
-ExperimentPlural = "experiments"
-
-class Experiment(launch_crd.K8sCR):
-  def __init__(self, version="v1alpha3", client=None):
-    super(Experiment, self).__init__(ExperimentGroup, ExperimentPlural, version, client)
-
-  def is_expected_conditions(self, inst, expected_conditions):
-    conditions = inst.get('status', {}).get("conditions")
-    if not conditions:
-      return False, ""
-    if conditions[-1]["type"] in expected_conditions:
-      return True, conditions[-1]["type"]
+import time
+
+from kubernetes.client import V1ObjectMeta
+
+from kubeflow.katib import KatibClient
+from kubeflow.katib import ApiClient
+from kubeflow.katib import V1beta1Experiment
+
+logger = logging.getLogger()
+logging.basicConfig(level=logging.INFO)
+
+FINISH_CONDITIONS = ["Succeeded", "Failed"]
+
+
+class JSONObject(object):
+    """ This class is needed to deserialize input JSON.
+    Katib API client expects JSON under .data attribute.
+    """
+
+    def __init__(self, json):
+        self.data = json
+
+
+def wait_experiment_finish(katib_client, experiment, timeout):
+    polling_interval = datetime.timedelta(seconds=30)
+    end_time = datetime.datetime.now() + datetime.timedelta(minutes=timeout)
+    experiment_name = experiment.metadata.name
+    experiment_namespace = experiment.metadata.namespace
+    while True:
+        try:
```

Comment:
```suggestion
        current_status = None
        try:
```
This could be one way to prevent that `current_status ` is undefined in case an exception is thrown.

Reply:
Make sense, thanks @thomas-mueller!